gen/internals.py

```diff
@@ -165,10 +172,16 @@ def __repr__(self):
 
 
 class Target:
-
+    """ Abstraction for specifying the final set of arguments and how
+    they conditions in which they are required
+    """
     # TODO(cmaloney): Make a better API for working with and managing sub scopes. The current
     # dictionary of dictionaries is really hard to use right.
     def __init__(self, variables=None, sub_scopes=None):
+        """
+        Variables: set of parameters
+        subscopes: mapping of variables to scopes
+        """
```

Comment:
Type annotations would be a better way to document these details. https://docs.python.org/3/library/typing.html